benchmarks/tensors/dibaryon/baryon_generator.cpp

```diff
@@ -230,40 +554,98 @@ void generate_function(std::string name)
-    O_i_update.store_in(&buf_O_i, {t, n, jCprime, jSprime, kCprime, kSprime, iCprime, iSprime, x, y});
-    P_r_update.store_in(&buf_P_r, {t, n, jCprime, jSprime, kCprime, kSprime, iCprime, iSprime, x, y});
-    P_i_update.store_in(&buf_P_i, {t, n, jCprime, jSprime, kCprime, kSprime, iCprime, iSprime, x, y});
 
     Bsingle_r_init.store_in(&buf_Bsingle_r);
     Bsingle_i_init.store_in(&buf_Bsingle_i);
 
-    Bsingle_r_update.store_in(&buf_Bsingle_r, {t, n, iCprime, iSprime, jCprime, jSprime, kCprime, kSprime, x, x2});
-    Bsingle_i_update.store_in(&buf_Bsingle_i, {t, n, iCprime, iSprime, jCprime, jSprime, kCprime, kSprime, x, x2});
-
     Bdouble_r_init.store_in(&buf_Bdouble_r);
     Bdouble_i_init.store_in(&buf_Bdouble_i);
-    Bdouble_r_update0.store_in(&buf_Bdouble_r, {t, n, iCprime, iSprime, jCprime, jSprime, kCprime, kSprime, x, x2});
-    Bdouble_i_update0.store_in(&buf_Bdouble_i, {t, n, iCprime, iSprime, jCprime, jSprime, kCprime, kSprime, x, x2});
-    Bdouble_r_update1.store_in(&buf_Bdouble_r, {t, n, iCprime, iSprime, jCprime, jSprime, kCprime, kSprime, x, x2});
-    Bdouble_i_update1.store_in(&buf_Bdouble_i, {t, n, iCprime, iSprime, jCprime, jSprime, kCprime, kSprime, x, x2});
+
+    buffer *q_r_bufs[Q_BLOCK_SIZE];
+    buffer *q_i_bufs[Q_BLOCK_SIZE];
+    buffer *o_r_bufs[O_BLOCK_SIZE];
+    buffer *o_i_bufs[O_BLOCK_SIZE];
+    buffer *p_r_bufs[P_BLOCK_SIZE];
+    buffer *p_i_bufs[P_BLOCK_SIZE];
+
+    for (int i = 0; i < Q_BLOCK_SIZE; i++) {
+      allocate_complex_buffers(
+          q_r_bufs[i], q_i_bufs[i],
+          { Vsnk }, str_fmt("buf_q_%d", i));
+    }
+    for (int i = 0; i < O_BLOCK_SIZE; i++) {
+      allocate_complex_buffers(
+          o_r_bufs[i], o_i_bufs[i],
+          { Vsnk }, str_fmt("buf_o_%d", i));
+    }
+    for (int i = 0; i < P_BLOCK_SIZE; i++) {
+      allocate_complex_buffers(
+          p_r_bufs[i], p_i_bufs[i],
+          { Vsnk }, str_fmt("buf_p_%d", i));
+    }
+
+    for (auto edges : blocked_q2userEdges) {
+      for (int i = 0; i < edges.size(); i++) {
+        auto edge = edges[i];
+        edge.q_r->store_in(q_r_bufs[i], {y});
+        edge.q_i->store_in(q_i_bufs[i], {y});
+      }
+    }
+
+    for (auto computations: Bsingle_updates) {
+      computation *real;
+      computation *imag;
+      std::tie(real, imag) = computations;
+      real->store_in(&buf_Bsingle_r, {t, n, iCprime, iSprime, jCprime, jSprime, kCprime, kSprime, x, x2});
+      imag->store_in(&buf_Bsingle_i, {t, n, iCprime, iSprime, jCprime, jSprime, kCprime, kSprime, x, x2});
+    }
+    for (auto computations: Blocal_updates) {
+      computation *real;
+      computation *imag;
+      std::tie(real, imag) = computations;
+      real->store_in(&buf_Blocal_r, {t, n, iCprime, iSprime, jCprime, jSprime, kCprime, kSprime, x});
+      imag->store_in(&buf_Blocal_i, {t, n, iCprime, iSprime, jCprime, jSprime, kCprime, kSprime, x});
+    }
+
+    for (auto edges : blocked_o2userEdges) {
```

Comment:
Move this and the next to put it just after the block doing

edge.q_r->store_in(q_r_bufs[i], {y});

This will make all the p, q and o store_in() call together.